controller.rst

```diff
@@ -244,41 +237,136 @@ The Symfony templating system and Twig are explained more in the
+    }
 
-What other services exist? To list all services, use the ``debug:container``
-console command:
+Awesome!
+
+What other services exist? Each page of the documentation will reveal more and more
+services you can use. To list *all* services, use the ``debug:container`` console
+command:
 
 .. code-block:: terminal
 
     $ php bin/console debug:container
 
-For more information, see the :doc:`/service_container` article.
+If need to control the *exact* value of an argument, you can override your controller's
+service config:
 
-.. tip::
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            # ...
+
+            # explicitly configure the service
+            AppBundle\Controller\LuckyController:
+                public: true
+                tags:
+                    # add multiple tags to controller multiple args
+                    - name: controller.service_arguments
+                      action: numberAction
+                      argument: logger
+                      # pass this specific service id
+                      id: monolog.logger.doctrine
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <!-- ... -->
+
+                <!-- Explicitly configure the service -->
+                <service id="AppBundle\Controller\LuckyController" public="true">
+                    <tag
+                        name="controller.service_arguments"
+                        action="numberAction"
+                        argument="logger"
+                        id="monolog.logger.doctrine"
+                    />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use AppBundle\Controller\LuckyController;
+        
+        $container->register(LuckyController::class)
+            ->setPublic(true)
+            ->addTag('controller.service_arguments', [
+                'action' => 'numberAction',
+                'argument' => 'logger',
+                'id' => 'monolog.logger.doctrine',
+            ])
+        ;
+
+For more information about services, see the :doc:`/service_container` article.
+
+.. note::
+    If this isn't working, make sure your controller is registered as a service,
+    :ref:`autoconfigured <services-autoconfigure>` and extends either
+    :class:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller` or
+    :class:`Symfony\\Bundle\\FrameworkBundle\\Controller\\AbstractController`. Or,
+    you can tag your service manually with ``controller.service_arguments``.
```

Comment:
I would replace this by `if you want to inject dependencies using arguments, your service has to be tagged with controller.service_arguments (already done in the standard edition).`